clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1201,13 +1201,26 @@ private ClusterAndWaitTime waitOnMetadata(String topic, Integer partition, long
             metadata.maybeThrowExceptionForTopic(topic);
             remainingWaitMs = maxWaitMs - elapsed;
             partitionsCount = cluster.partitionCountForTopic(topic);
-        } while (partitionsCount == null || (partition != null && partition >= partitionsCount));
+
+            if (!topicPartitionMetadataFound(partitionsCount, partition)) {
+                String expectedMetadataNotFoundMessage = partitionsCount == null ?
+                    String.format("Topic %s not present in metadata after %d ms. Retrying until %d ms elapses",
+                        topic, elapsed, maxWaitMs) :
+                    String.format("Partition %d of topic %s with partition count %d is not present in metadata after %d ms. Retrying until %d ms elapses",
+                        partition, topic, partitionsCount, elapsed, maxWaitMs);
+                log.info(expectedMetadataNotFoundMessage);
```

Comment:
could we set this at debug level or lower? I imagine it could be a bit spammy.

Reply:
I understand that this can get a bit spammy at times, but I decided to make it a info log since making it a debug log will still require one to enable debug logs here and based on what I have seen, from a connect point of view, enabling debug logs in KafkaProducer is not one of the top things one tries and thus would still require some digging before one reaches here in troubleshooting

Reply:
Right -- but this would affect every producer and not just connect. 😅 

I would think that if connect was aware of this issue, they could share this tip for debugging.

Reply:
I have updated the same. Thanks